File: Julia/Path.jl
#! /usr/bin/env julia

println("""
ホームディレクトリ:       $(homedir())
カレントディレクトリ:     $(pwd())

このファイルのフルパス:   $(@__FILE__)
このファイルの実行パス:   $(PROGRAM_FILE)
このファイルの名前:       $(basename(@__FILE__))
このファイルのある場所:   $(@__DIR__)
""")

File: JavaScript/Path.js
#! /usr/bin/env node

const process = require("process");
const os = require("os");

console.log(`
ホームディレクトリ:       ${os.homedir()}
カレントディレクトリ:     ${process.cwd()}

このファイルのある場所:   ${__dirname}
`)